packages/core/src/browser/tree/tree.ts

```diff
@@ -329,4 +356,29 @@ export class TreeImpl implements Tree {
         }
     }
 
+    async markAsBusy(raw: TreeNode, ms: number, token: CancellationToken): Promise<void> {
+        const node = this.validateNode(raw);
+        if (node) {
+            await this.markAsBusy(node, ms, token);
+        }
+    }
+    protected async doMarkAsBusy(node: Mutable<TreeNode>, ms: number, token: CancellationToken): Promise<void> {
+        try {
+            await timeout(ms, token);
+            this.doSetBusy(node, true);
```

Comment:
exit early here if `token.isCancellationRequested()` is truish, no?

Reply:
yes, it makes sense

Reply:
Ah! Now I got it, it's the `cancelled` exception which will be thrown from `timeout` in that case, thus it won't be set `busy`. 
That wasn't that obvious to me on first scan.

Reply:
maybe a short comment at `timeout`'s def would be fine.

Reply:
ok, updated docs